tvx/builders/generator.go

```diff
@@ -0,0 +1,166 @@
+package builders
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sync"
+
+	"github.com/filecoin-project/oni/tvx/schema"
+)
+
+// Generator is a batch generator and organizer of test vectors.
+//
+// Test vector scripts are simple programs (main function). Test vector scripts
+// can delegate to the Generator to handle the execution, reporting and capture
+// of emitted test vectors into files.
+//
+// Generator supports the following CLI flags:
+//
+//  -o <directory>
+//		directory where test vector JSON files will be saved.
+//
+//  -f <regex>
+//		regex filter to select a subset of vectors to execute; matched against
+//	 	the vector's ID.
+//
+// Scripts can bundle test vectors into "groups". The generator will execute
+// each group in parallel, and will write each vector in a file:
+// <output_dir>/<group>-<sequential>.json
```

Comment:
```suggestion
// <output_dir>/<group>-<vector_ID>.json
```

Might be more stable and not cause problems when re-generating vectors over existing files (easier to identify also).

Reply:
Done!